app/controllers/application_controller.rb

```diff
@@ -1,10 +1,149 @@
-require './config/environment'
-
 class ApplicationController < Sinatra::Base
-
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if session[:id]
+      redirect '/tweets'
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if empty_fields?
+      redirect '/signup'
+    else
+      @user = User.create(params)
+      session[:id] = @user.id
+      redirect '/tweets'
+    end
+  end
+
+  get '/tweets' do
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).